File: mvvmbase/src/main/java/com/bing/mvvmbase/base/viewpager/BaseFragmentStatePagerAdapter.java
package com.bing.mvvmbase.base.viewpager;

import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

public class BaseFragmentStatePagerAdapter<E> extends FragmentStatePagerAdapter {
	protected List<E> mData;

	public BaseFragmentStatePagerAdapter(FragmentManager fm, List<E> data) {
		super(fm);
		mData = data;
	}

	@Override
	public Fragment getItem(int position) {
		return null;
	}

	@Override
	public int getCount() {
		return mData.size();
	}
}

File: mvvmbase/src/main/java/com/bing/mvvmbase/base/viewpager/BaseFragmentStatePagerAdapter.kt
package com.bing.mvvmbase.base.viewpager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class BaseFragmentStatePagerAdapter<E>(fm: FragmentManager, protected var mData: List<E>) : FragmentStatePagerAdapter(fm) {

        override fun getItem(position: Int): Fragment? {
                return null
        }

        override fun getCount(): Int {
                return mData.size
        }
}
